### PR TITLE
⚠️  Add go modules to IPAM api

### PR DESCRIPTION
**Pull request by furkatgofurov7**
**What this PR does / why we need it**:
This PR adds go modules to IPAM API. The reason to add the go modules in API is to uplift IPAM for CAPM3. List of modules updated:

- sigs.k8s.io/controller runtime from `v0.7.0` to `v0.9.0-beta.0`
- sigs.k8s.io/cluster-api from `v0.3.12` to `v0.3.11-0.20210429135050-9d825629d89c` [v1alpha4](https://pkg.go.dev/sigs.k8s.io/cluster-api@v0.3.11-0.20210429135050-9d825629d89c/api/v1alpha4)
- k8s.io/client-go from `v1.5.1` to `v0.21.0`
- k8s.io/apimachinery from `v0.20.1` to `v0.21.0`
- k8s.io/apiextensions-apiserver from `v0.20.1` to `v0.21.0`
- k8s.io/api from `v0.20.1` to `v0.21.0`

**Important:**
Once this PR goes in, those who need to uplift IPAM, and they are only using IPAM APIs , you need to change your go modules to use

```
github.com/metal3-io/ip-address-manager/api
```
 instead of:
```
github.com/metal3-io/ip-address-manager
```

If you are using/referring to other parts of code in IPAM not residing in /api, you need to use both the modules:
```
github.com/metal3-io/ip-address-manager/api
github.com/metal3-io/ip-address-manager
```

### CODE REVIEW

**Comment by fmuyassarov** [comment]
Seems we can't run integration test against this branch

**Comment by furkatgofurov7** [comment]
> Seems we can't run integration test against this branch

Yes, I was checking it in project-infra, and this is because we configured to run only in PRs' going into [master](https://github.com/metal3-io/ip-address-manager/pull/32) IMO, and we could  also specify other branch than master in configuration if needed.

File: Makefile [hunk 1]
```diff
@@ -170,9 +171,9 @@ generate: ## Generate code
 .PHONY: generate-go
 generate-go: $(CONTROLLER_GEN) $(MOCKGEN) $(CONVERSION_GEN) $(KUBEBUILDER) $(KUSTOMIZE) ## Runs Go related generate targets
 	go generate ./...
```
Comment (maelk) [inline Makefile:173]: this needs to be done for api folder too

Comment (maelk) [reply Makefile:173]: line 163 also

Comment (maelk) [reply Makefile:173]: the lint target in line 140 also and the lint-full also

**Comment by furkatgofurov7** [comment]
This is pointing to wrong branch (capi-v1alpah4), will open towards master. 

/close

**Pull request closed without merging**